pkg/scheduler/framework/plugins/registry.go

```diff
@@ -45,33 +49,59 @@ import (
 // through the WithFrameworkOutOfTreeRegistry option.
 func NewInTreeRegistry() runtime.Registry {
 	return runtime.Registry{
-		selectorspread.Name:                        selectorspread.New,
-		imagelocality.Name:                         imagelocality.New,
-		tainttoleration.Name:                       tainttoleration.New,
-		nodename.Name:                              nodename.New,
-		nodeports.Name:                             nodeports.New,
-		nodepreferavoidpods.Name:                   nodepreferavoidpods.New,
-		nodeaffinity.Name:                          nodeaffinity.New,
-		podtopologyspread.Name:                     podtopologyspread.New,
-		nodeunschedulable.Name:                     nodeunschedulable.New,
-		noderesources.FitName:                      noderesources.NewFit,
-		noderesources.BalancedAllocationName:       noderesources.NewBalancedAllocation,
-		noderesources.MostAllocatedName:            noderesources.NewMostAllocated,
-		noderesources.LeastAllocatedName:           noderesources.NewLeastAllocated,
-		noderesources.RequestedToCapacityRatioName: noderesources.NewRequestedToCapacityRatio,
-		volumebinding.Name:                         volumebinding.New,
-		volumerestrictions.Name:                    volumerestrictions.New,
-		volumezone.Name:                            volumezone.New,
-		nodevolumelimits.CSIName:                   nodevolumelimits.NewCSI,
-		nodevolumelimits.EBSName:                   nodevolumelimits.NewEBS,
-		nodevolumelimits.GCEPDName:                 nodevolumelimits.NewGCEPD,
-		nodevolumelimits.AzureDiskName:             nodevolumelimits.NewAzureDisk,
-		nodevolumelimits.CinderName:                nodevolumelimits.NewCinder,
-		interpodaffinity.Name:                      interpodaffinity.New,
-		nodelabel.Name:                             nodelabel.New,
-		serviceaffinity.Name:                       serviceaffinity.New,
-		queuesort.Name:                             queuesort.New,
-		defaultbinder.Name:                         defaultbinder.New,
-		defaultpreemption.Name:                     defaultpreemption.New,
+		selectorspread.Name:      selectorspread.New,
+		imagelocality.Name:       imagelocality.New,
+		tainttoleration.Name:     tainttoleration.New,
+		nodename.Name:            nodename.New,
+		nodeports.Name:           nodeports.New,
+		nodepreferavoidpods.Name: nodepreferavoidpods.New,
+		nodeaffinity.Name:        nodeaffinity.New,
+		podtopologyspread.Name:   podtopologyspread.New,
+		nodeunschedulable.Name:   nodeunschedulable.New,
+		noderesources.FitName: func(plArgs apiruntime.Object, fh framework.Handle, opts ...framework.PluginOptionFunc) (framework.Plugin, error) {
```

Comment:
If we are going to wrap anyways, we should keep the same signature for `PluginFactory`, as callers of the Registry are not going to pass any option.

Reply:
+1